compiler/circle-mpqsolver/src/pattern/PatternResolver.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PatternResolver.h"
+
+#include <luci/IR/CircleNodes.h>
+
+using namespace mpqsolver::pattern;
+
+bool Q8LayerNormWithQ16VarianceResolver::resolve(luci::Module *module,
+                                                 std::map<std::string, LayerParam> &nodes_params)
+{
+  if (!module)
+  {
+    return false;
+  }
+
+  for (size_t idx = 0; idx < module->size(); ++idx)
+  {
+    auto graph = module->graph(idx);
+
+    for (auto node : loco::active_nodes(loco::output_nodes(graph)))
+    {
+      auto const final_mul = dynamic_cast<luci::CircleMul *>(node);
```

Comment:
Could you make this pattern as a class? (ex: see FuseInstanceNormPass.cpp) There can be multiple patterns for layernorm Op. It is difficult to add new patterns to the current code.

Reply:
Ahhh. Ok.

Reply:
Please write comments about the pattern like `FuseInstanceNormPass`.

From FuseInstanceNormPass.cpp.
```cpp
 *                 [In]
 *                   |
 *                   V
 *     +----------- ifm -----+   (reduction indicies)
 *     |             |       |       |
 *     |             |       V       V
 *     |             |      mean_of_ifm ----------------+
 *     |             V       |                          |
 *     |           sqdiff <--+   (reduction indicies)   |
 *     |             |             |                    |
 *     |             V             |                    |
 *     |      mean_as_variance <---+  const_as_epsilon  |
 *     |             |                 |                |
 *     |             V                 |                |
 *     |      add_as_variance <--------+                |
 *     |             |                                  |
 *     |             V                                  |
 *     |           rsqrt   const_as_gamma               |
 *     |             |        |                         |
 *     |             V        |                         |
 *     |         mul_gamma <--+                         |
 *     |          |     |                               |
 *     V          V     V                               |
 * mul_as_scaled_ifm   mul_as_scaled_mean <-------------+
 *         |                   |
 *         |   const_as_beta   |
 *         |         |         V
 *         |         +------> sub
 *         V                   |
 *  add_as_terminal <----------+
 *         |
 *         V
 *       [Out]
```

And we need a unittest like FuseInstanceNormPass.test.cpp.

Reply:
ok

Reply:
@jinevening
done